### PR TITLE
MQE-2142: Magento 2.4.0 MFTF BiC list

### PR DESCRIPTION
**Pull request by jilu1**
## Purpose of this pull request

The PR contains MFTF Tests backward incompatible changes between 2.3.0 and 2.4.0-develop. 
This the the result of MFTF SVC tool with some post processing.
This documents can be added in any folder appropriate and linked into main BiC page.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by jilu1** [comment]
I accidentally close the previous PR and could not reopen. Here is the new PR.

**Comment by jilu1** [comment]
> There is extra dash (`-`) on lines containing a `how` parameter.
> Should be:
> 
> ```
>   - what: Magento_SharedCatalog/Section/AdminSharedCatalogsSection/multicheck
>     how: <section> <element> was removed
>   - what: Magento_SharedCatalog/Section/AdminSharedCatalogsSection/selectAll
>     how: <section> <element> was removed
> ```

Fixed

**Comment by jilu1** [comment]
> There is an extra space there now. YAML is sensitive to indentation.

Removed

File: src/_data/toc/mftf.yml [hunk 1]
```diff
@@ -87,8 +87,7 @@ pages:
     children:
 
       - label: Action group reference
-        url: /guides/v2.3/reference/mftf/action-groups.html
-        versionless: true
+        url: /reference/mftf/action-groups.html
```
Comment (dshevtsov) [inline src/_data/toc/mftf.yml:90]: Making the TOC item to appear in both 2.3. and 2.4 versions of the site.

File: src/_data/toc/mftf.yml [hunk 2]
```diff
@@ -99,8 +98,7 @@ pages:
         versionless: true
       
       - label: MFTF functional test reference
-        url: /guides/v2.3/reference/mftf/functional-tests.html
-        versionless: true
+        url: /reference/mftf/functional-tests.html
```
Comment (dshevtsov) [inline src/_data/toc/mftf.yml:101]: Making the TOC item to appear in both 2.3. and 2.4 versions of the site.

**Comment by dshevtsov** [comment]
To consider in the future: it would make sense to group data by a type of change and present it as lists instead of tables to make the content more comfortable to read.

Example:

>  Entity <data> element was removed at:
>  - Magento_Staging/Data/S5UpdateNameStaging/startDate
>  - Magento_Staging/Data/S5UpdateNameStaging/endDate
>  - Magento_Staging/Data/S5UpdateNameStaging/productPrice
>  - Magento_Staging/Data/S5UpdateNameStaging/scheduleStartDate
>  - Magento_Staging/Data/S5UpdateNameStaging/scheduleEndDate
>

**Comment by jilu1** [comment]
@dshevtsov Please let me know if there is anything from my side. Thanks!

**Comment by dshevtsov** [comment]
@jilu1, I'm good if you don't have concerns or remarks related to my changes.
Waiting for approval from @dobooth as an owner of the MFTF documentation on devdocs.

**Comment by dshevtsov** [comment]
running tests

**Pull request merged**